sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileSystemAPITest.groovy

```diff
@@ -838,7 +1010,175 @@ class FileSystemAPITest extends APISpec {
+        "control"    | "disposition"      | "encoding"      | "language"      | null       | "type"
+    }
+
+    def "Create if not exists file options with metadata"() {
+        setup:
+        def metadata = new HashMap<String, String>()
+        if (key1 != null && value1 != null) {
+            metadata.put(key1, value1)
+        }
+        if (key2 != null && value2 != null) {
+            metadata.put(key2, value2)
+        }
+        def options = new DataLakePathCreateOptions().setMetadata(metadata)
+        def result = fsc.createFileIfNotExistsWithResponse(generatePathName(), options, null, null)
+
+        expect:
+        result.getStatusCode() == 201
+        def properties = fsc.getProperties()
+        // Directory adds a directory metadata value
+        for(String k : metadata.keySet()) {
+            properties.getMetadata().containsKey(k)
+            properties.getMetadata().get(k) == metadata.get(k)
+        }
+
+        where:
+        key1  | value1 | key2   | value2
+        null  | null   | null   | null
+        "foo" | "bar"  | "fizz" | "buzz"
+    }
+
+    def "Create if not exists file options with permissions and umask"() {
+        setup:
+        def permissions = "0777"
+        def umask = "0057"
+        def options = new DataLakePathCreateOptions().setPermissions(permissions).setUmask(umask)
+        def result = fsc.createFileIfNotExistsWithResponse(generatePathName(), options, null, null).getValue()
+
+        when:
+        def acl = result.getAccessControlWithResponse(true, null, null, null).getValue()
+
+        then:
+        PathPermissions.parseSymbolic("rwx-w----").toString() == acl.getPermissions().toString()
+    }
+
+    def "Create if not exists file options with lease id"() {
+        when:
+        def leaseId = UUID.randomUUID().toString()
+        def options = new DataLakePathCreateOptions().setProposedLeaseId(leaseId).setLeaseDuration(15)
+        def response = fsc.createFileIfNotExistsWithResponse(generatePathName(), options, null, null)
+
+        then:
+        response.getStatusCode() == 201
+    }
+
+    def "Create if not exists file options with lease id error"() {
+        when:
+        def leaseId = UUID.randomUUID().toString()
+        def options = new DataLakePathCreateOptions().setProposedLeaseId(leaseId)
+        fsc.createFileIfNotExistsWithResponse(generatePathName(), options, null, null)
+
+        then:
+        // lease duration must also be set, or else exception is thrown
+        thrown(DataLakeStorageException)
+    }
+
+    def "Create if not exists file options with lease duration"() {
+        when:
+        def leaseId = UUID.randomUUID().toString()
+        def options = new DataLakePathCreateOptions().setLeaseDuration(15).setProposedLeaseId(leaseId)
+        def response = fsc.createFileIfNotExistsWithResponse(generatePathName(), options, null, null)
+
+        then:
+        response.getStatusCode() == 201
```

Comment:
should we check if lease has been acquired?

Reply:
This is how we checked in .NET - https://github.com/Azure/azure-sdk-for-net/blob/ff1ef6a4bbc0286fbaebbc458ce569ce85f7d9b1/sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs#L513